packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -1936,8 +2798,250 @@ function revisionFromRevInfos(
 	return revInfos[0].revision;
 }
 
-function getActiveFieldChanges(changes: ModularChangeset): FieldChangeMap {
-	return (changes.constraintViolationCount ?? 0) === 0
-		? changes.fieldChanges
-		: new Map<FieldKey, FieldChange>();
+function mergeBTrees<K, V>(tree1: BTree<K, V>, tree2: BTree<K, V>): BTree<K, V> {
+	const result = tree1.clone();
+	tree2.forEachPair((k, v) => {
+		result.set(k, v);
+	});
+
+	return result;
+}
+
+function mergeNestedMaps<K1, K2, V>(
+	map1: NestedMap<K1, K2, V>,
+	map2: NestedMap<K1, K2, V>,
+): NestedMap<K1, K2, V> {
+	const merged: NestedMap<K1, K2, V> = new Map();
+	populateNestedMap(map1, merged, true);
+	populateNestedMap(map2, merged, true);
```

Comment:
I'm concerned that map1 and map2 may have some keys in common, in which case the data for map1 will be lost.

Reply:
As discussed: this is fine because any overwritten data will be re-discovered through the top-down compose process.